oss-internship-2020/libtiff/test/defer_strile_writing.cc

```diff
@@ -0,0 +1,313 @@
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFSetFieldU1 fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "Can't set SamplesPerPixel tag";
+
+  status_or_int =
+      api.TIFFSetFieldU1(&tif, TIFFTAG_PLANARCONFIG, PLANARCONFIG_CONTIG);
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFSetFieldU1 fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "Can't set PlanarConfiguration tag";
+
+  status_or_int = api.TIFFSetFieldU1(&tif, TIFFTAG_ROWSPERSTRIP, kRowsPerStrip);
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFSetFieldU1 fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue()) << "Can't set RowsPerStrip tag";
+
+  status_or_int = api.TIFFDeferStrileArrayWriting(&tif);
+  ASSERT_THAT(status_or_int, IsOk())
+      << "TIFFDeferStrileArrayWriting fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFDeferStrileArrayWriting return unexpected value";
+
+  status_or_int = api.TIFFWriteCheck(&tif, 0, test_var.PtrBefore());
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFWriteCheck fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFWriteCheck return unexpected value";
+
+  status_or_int = api.TIFFWriteDirectory(&tif);
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFWriteDirectory fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFWriteDirectory return unexpected value";
+
+  // Force writing of strile arrays
+  status_or_int = api.TIFFSetDirectory(&tif, 0);
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFSetDirectory fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFSetDirectory return unexpected value";
+
+  status_or_int = api.TIFFForceStrileArrayWriting(&tif);
+  ASSERT_THAT(status_or_int, IsOk())
+      << "TIFFForceStrileArrayWriting fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFForceStrileArrayWriting return unexpected value";
+
+  status_or_int = api.TIFFSetDirectory(&tif, 1);
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFSetDirectory fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFSetDirectory return unexpected value";
+
+  status_or_int = api.TIFFForceStrileArrayWriting(&tif);
+  ASSERT_THAT(status_or_int, IsOk())
+      << "TIFFForceStrileArrayWriting fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFForceStrileArrayWriting return unexpected value";
+
+  // Now write data on frist directory
+  status_or_int = api.TIFFSetDirectory(&tif, 0);
+  ASSERT_THAT(status_or_int, IsOk()) << "TIFFSetDirectory fatal error";
+  EXPECT_THAT(status_or_int.value(), IsTrue())
+      << "TIFFSetDirectory return unexpected value";
+
+  if (tiled) {
+    for (int i = 0; i < (height + 15) / 16; ++i) {
+      std::array<unsigned char, kTileBufferSize> tilebuffer;
+      tilebuffer.fill(i);
+      sapi::v::Array<unsigned char> tilebuffer_(tilebuffer.data(),
+                                                kTileBufferSize);
```

Comment:
I'd go with `tilebuffer.size()` here (so that it is clear that the length parameter won't be bigger than tilebuffer without looking 2 lines up) but this is probably only a preference - so it is fine the way it is.